node/src/components/consensus/highway_core/state.rs

```diff
@@ -745,7 +748,31 @@ impl<C: Context> State<C> {
         // A block at height > 0 always has at least its parent entry in skip_idx.
         #[allow(clippy::integer_arithmetic)]
         let i = max_i.min(block.skip_idx.len() - 1);
-        self.find_ancestor(&block.skip_idx[i], height)
+        self.find_ancestor_proposal(&block.skip_idx[i], height)
+    }
+
+    /// Returns the ancestor of the unit with the given `hash`, on the specified `height`, or
+    /// `None` if the unit's height is lower than that.
+    pub(crate) fn find_ancestor_unit<'a>(
```

Comment:
It would be awesome to deduplicate `find_ancestor_proposal` and `find_ancestor_unit`, as the logic is essentially the same, just on different types, but it doesn't have to be in this PR.

Reply:
I agree but I couldn't think of an implementation that makes sense. Do you have any suggestions?

Reply:
I guess we'd need a trait like `ReadByHash<T>` implemented twice for `State`, and another trait (`VertexLike` or something) containing methods like `hash`, `height` and `skip_idx` implemented for blocks and units, then we could have `find_ancestor<T>() where State: ReadByHash<T>, T: VertexLike`...
Probably too much effort for relatively little gain :grimacing:

Reply:
Is this method equivalent to `find_in_swimlane`?

The Highway code uses the word "sequence number" for this elsewhere, to distinguish it from block height or validator index.